presto-main/src/main/java/com/facebook/presto/execution/DropBranchTask.java

```
@@ -0,0 +1,77 @@
+import com.facebook.presto.spi.security.AccessControl;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.tree.DropBranch;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.transaction.TransactionManager;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.createQualifiedObjectName;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_FOUND;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MISSING_TABLE;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+
+public class DropBranchTask
+        implements DDLDefinitionTask<DropBranch>
+{
+    @Override
+    public String getName()
+    {
+        return "DROP BRANCH";
+    }
+
+    @Override
+    public ListenableFuture<?> execute(DropBranch statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, Session session, List<Expression> parameters, WarningCollector warningCollector)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
+        Optional<TableHandle> tableHandleOptional = metadata.getMetadataResolver(session).getTableHandle(tableName);
+
+        if (!tableHandleOptional.isPresent()) {
+            if (!statement.isTableExists()) {
+                throw new SemanticException(MISSING_TABLE, statement, "Table '%s' does not exist", tableName);
+            }
+            return immediateFuture(null);
+        }
+
+        Optional<MaterializedViewDefinition> optionalMaterializedView = metadata.getMetadataResolver(session).getMaterializedView(tableName);
+        if (optionalMaterializedView.isPresent()) {
+            if (!statement.isTableExists()) {
```

should we throw an error in cases when `if exists` is present in the query and it is a materialized view? If drop branch is just not supported for MVs, ideally error should be thrown irrespective of exists check?

---

Alternate way of handling this can be introducing an enum Type in DropBranch and have only TABLE as the supported type.